### DISCUSSION TITLE
How to use Walker as clipboard manager?

### DISCUSSION THREAD

**Discussion by TheFrankyDoll** [abenz1267/walker#36] [Q&A] [answered]
Readme mentions clipboard with fuzzy find and images as a builtin module. What command should I use to show clipboard with walker? 
There seems to be no information on how to do it.

**Comment by abenz1267** [comment]
You just add it to your config like any other module. No magic involved. It only works with wl-clipboard, though. As mentioned in the readme. Walker needs to run as a service.

**Reply by TheFrankyDoll** [reply]
I do use wl-clipboard as part of my NixOS setup. So I tried to add
```
        // ...
"modules": [
	{
		"name": "clipboard",
		"prefix": "",
		"switcher_exclusive": true
	}
        // ...
```
but it doesn't seem to create a switcher option, nor I see anything from what I copied if I don't put `"switcher_exclusive": true` to the list.

Also, I can't find any information on how to properly start Walker as a service, is there a special flag I need to run it with, or should I create a systemd service?

I definitely missing something - is there any example I could use as a reference? I would suggest adding this in FAQ too, as I feel it could confuse more new users like myself.

**Reply by abenz1267** [reply, marked as answer]
If you don't start Walker as a service, the clipboard module won't be loaded. Check the readme. You can create a systemd service or if using hyprland use "exec-once" in your hyprland config to start the service.

**Reply by LeonardoMor** [reply]
The README mentions the `:` prefix, but if I open Walker and type `:` and then start typing things that I thing are on the clipboard, then it gives me options to do a web search, it doesn't show items in the clipboard.

In general how do prefixes work? Did I misunderstood?

Also how do I run arbitrary shell commands?

**Reply by abenz1267** [reply]
did you install the elephant clipboard provider?

While walker has things preconfigured for providers, it'll ignore configs for providers that arent installed.

**Reply by ngocphamm** [reply]
Seeing this kinda behavior on Ubuntu 25.04. @abenz1267 I guess it needs to start as a service like you said above (but over a year ago already so not sure if it's still relevant).

For Ubuntu, does that mean I need to create a service with systemd, to execute the command `walker --gapplication-service`?

**Reply by abenz1267** [reply]
Walker 1.0.0 does not need to be started as service in order to use any provider. I'd still highly recommend it for performance-reasons.

You can just create an autostart entry in `~/.config/autostart`.

**Reply by ngocphamm** [reply]
Makes sense! I ran `walker --gapplication-service &` first then from that point using a keyboard shortcut to trigger `walker` and it's working well for me.

However, I think the clipboard provider doesn't seemt to work. I do have `wl-clipboard` installed. Is there any other external libraries needed for it to work?

**Reply by abenz1267** [reply]
no, really just wl-clipboard. but most likely that's interferring with anything that ubuntu has installed, as you can attach multiple clipboards afaik.

what makes you think it's not working? is the clipboard module not available? is it empty?

**Reply by ngocphamm** [reply]
Triggering walker and type `:` and it just shows `No Results`, even though I just copied a bunch of stuff. I do have clipboard module with elephant.

<img width="1173" height="315" alt="image" src="https://github.com/user-attachments/assets/f81cbb37-b661-4f7c-b7f9-05c63016ef2f" />

I think you might be right that it's interferring with something else on my system. I have a Gnome Extension called [Clipboard Indicator](https://extensions.gnome.org/extension/779/clipboard-indicator/) for managing the clipboard. I just disabled that and walker clipboard still doesn't show anything (I copied something after disabling that gnome extension).

**Reply by abenz1267** [reply]
what does `elephant` output on start?

**Reply by abenz1267** [reply]
wait. yeah. no results means that indeed wl-clipboard is not working. as i expected.

**Reply by ngocphamm** [reply]
<img width="1288" height="736" alt="image" src="https://github.com/user-attachments/assets/6b8c53ea-fb34-458c-90d6-75a2fb5efa78" />

How do I verify that `wl-cliboard` is working? I thought it is, because using walker, type `.` and search for emojis, and hit enter, and the emoji is copied to clipboard. :thinking:

**Reply by abenz1267** [reply]
can u do `wl-paste --watch echo 'test' ` in a terminal and just copy some text? does it print `test` everytime you copy something?

**Reply by ngocphamm** [reply]
I guess this is the issue?

```
~ »  wl-paste --watch echo 'test'
Watch mode requires a compositor that supports the wlroots data-control protocol
```

**Reply by abenz1267** [reply]
yes

**Reply by ngocphamm** [reply]
This sounds related, but I don't understand most of these, so I guess I will have to accept not being able to use walker as a clipboard manager under Ubuntu.

https://gitlab.gnome.org/GNOME/mutter/-/issues/524

**Reply by abenz1267** [reply]
as long as no one adds handling for whatever gnome is doing... no.

**Reply by LeonardoMor** [reply]
> did you install the elephant clipboard provider?
> 
> While walker has things preconfigured for providers, it'll ignore configs for providers that arent installed.

I did:

```output
❯ paru -Qqe | grep -Ei 'elephant.?clip' 
elephant-clipboard
```

Still, launching walker then typing `:` suggests a web search.

However, if I lauch walker and type `/` then select `clipboard`, that works as intended.

**Reply by abenz1267** [reply]
walker version?

**Reply by LeonardoMor** [reply]
```
❯ walker --version
Running Service: true
0.13.26
```

**Reply by abenz1267** [reply]
yeah, that wont work. you need walker 1.0.0.

**Reply by LeonardoMor** [reply]
Yes, thank you!
